chain/client/src/sync/state.rs

```diff
@@ -179,6 +179,44 @@ impl ExternalConnection {
             }
         }
     }
+
+    pub async fn list_state_parts(
+        &self,
+        shard_id: ShardId,
+        directory_location: &str,
+    ) -> Result<Vec<String>, anyhow::Error> {
+        let _timer = metrics::STATE_SYNC_DUMP_LIST_OBJECT_ELAPSED
+            .with_label_values(&[&shard_id.to_string()])
+            .start_timer();
+        match self {
+            ExternalConnection::S3 { bucket } => {
+                let prefix = format!("{}/", directory_location);
+                let list_results = bucket.list(prefix.clone(), Some("/".to_string())).await?;
+                tracing::debug!(target: "state_sync_dump", shard_id, ?directory_location, "List state parts in s3");
+                let mut file_names = vec![];
+                for res in list_results {
+                    for obj in res.contents {
+                        let file_name = obj.key.replace(&prefix, "");
```

Comment:
If `directory_location` doesn't start with `s3://` or similar,
then I'd suggest using `PathBuf::file_name()` instead of manipulating strings.

```
let path_buf = PathBuf::from(obj.key);
let file_name = path_buf.file_name().unwrap().to_str().unwrap();
```